### PR TITLE
Newsletter Accent Color: Display warning when accent color with too low contrast ratio is selected

### LINKED ISSUE #66697: Newsletter Accent Color: Display warning when in-accessible color selected
Image

- [ ] Display the warning if in-accessible color is selected
- [ ] Hide the warning once an accessible color is re-selected

Detailed designs: BcUdhLlhAp7UHtQHNTI4zA-fi-4163%3A112879

## Validating Color Contrast Correctness
We need to validate against the Lettre theme's background color. pNEWy-fia-p2
Make use of [the Gutenberg's `ColorChecker` component](https://href.li/?https://github.com/WordPress/gutenberg/blob/84b888735da778f3501610741f0d8bb3fda2884d/packages/block-editor/src/components/contrast-checker/README.md#development-guidelines) if possible.

## Related
Newsletter Accent Color: a11y pdDOJh-G0-p2

### PR DESCRIPTION
**Pull request by mashikag**
Previous to this PR when user selects an accent color they have no idea of how it will be displayed against the stite's background color. This PR implements a warning message that will be displayed if user selects an accent color where the color to site's background color ratio is lower than the minimum color ratio defined by WCAG standards (4.5:1). 

Screen Shot 2022-09-06 at 17 09 31

I used [this](https://dev.to/alvaromontoro/building-your-own-color-contrast-checker-4j7o) article as a guide to create the contrast checking utility functions.

#### Proposed Changes

* check for minimum contrast ratio between the selected accent color and the Lettre theme's bg color

#### Testing Instructions

* Go to http://calypso.localhost:3000/setup/newsletterSetup?flow=newsletter

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Resolves #66697

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx [hunk 1]
```diff
@@ -120,81 +127,93 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 	};
 
 	const stepContent = (
-		<>
```
Comment (mashikag) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx:123]: Yuck. It looks as if I touched every line in this block. All I did was remove the unnecessary fragment which resulted in one tab reduction for all encapsulated lines. I can revert this change if one prefers I did not do that.

Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx:123]: In these cases, I recommend to the reviewer to ignore whitespace (add `?w=1` to the URL). Not a problem.

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/utils.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+const colorContrastRatio = ( color1luminance: number, color2luminance: number ) =>
+	color1luminance > color2luminance
+		? ( color2luminance + 0.05 ) / ( color1luminance + 0.05 )
+		: ( color1luminance + 0.05 ) / ( color2luminance + 0.05 );
+
+const lettreThemeBgColorLuminance = luminance( LETTRE_THEME_SITE_BACKGROUND_COLOR_RGB );
+
+export const hasMinContrast = ( textColorRgb: RGB ): boolean => {
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/utils.ts]: Why not allow users to adjust?

```suggestion
export const hasMinContrast = ( textColorRgb: RGB, minContrast = MIN_CONTRAST_RATIO ): boolean => {
```

File: client/signup/icons/index.tsx [hunk 3]
```diff
@@ -109,6 +110,21 @@ export const tip: ReactElement = (
...
+			x1="6.99994"
+			y1="14.9816"
+			x2="12.9999"
+			y2="14.9816"
+			stroke={ colors[ 'Yellow 40' ] }
+			strokeWidth="1.3"
+		/>
+		<line x1="8" y1="17.35" x2="12" y2="17.35" stroke={ colors[ 'Yellow 40' ] } strokeWidth="1.3" />
+	</SVG>
+);
```
Comment (alshakero) [inline client/signup/icons/index.tsx]: This looks to be the same as `tip`. You can change the color in CSS if you want.

**Review by alshakero — CHANGES_REQUESTED** [review]
Love it. The `utils` file is a perfect candidate to live [here](https://github.com/Automattic/wp-calypso/blob/trunk/packages/onboarding/src/utils.ts). This way we can use it everywhere!

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/utils.ts (hunk 2 above)
Comment (mashikag) [reply client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/utils.ts]: I suppose no reason not to. 👍

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx (hunk 1 above)
Comment (mashikag) [reply client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx:123]: Omg... 😱  I never knew this. This is really useful. Thanks!

File: client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/style.scss [hunk 4]
```diff
@@ -72,6 +72,20 @@ $border-radius: 4px;
 			}
 		}
 
+		.newsletter-setup__contrast-warning {
+			display: flex;
+			font-size: $font-body-small;
+			margin-bottom: 32px;
+
+			.newsletter-setup__contrast-warning-icon-container {
+				margin-top: 4px;
```
Comment (yansern) [inline client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/style.scss:81]:
```suggestion
				margin-top: 4px;
				margin-right: 4px;
```

It just makes it look more breathable.

Before:
image

After: 
image

**Comment by yansern** [comment]
A quick smoke-test shows that it works well. Will leave it to @alshakero for final approval.

**Review by alshakero — APPROVED** [review]
The code looks great, and the functionality is perfect to spec.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7514767

Thank you @mashikag for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7514767).

**Pull request merged**